armi/reactor/blocks.py

```diff
@@ -1650,6 +1650,43 @@ def getDuctOP(self):
         duct = self.getComponent(Flags.DUCT, exact=True)
         return duct.getDimension("op")
 
+    def setB10VolParam(self, heightHot):
+        """Set the b.p.initialB10ComponentVol param according to the volume of boron-10 containing components."""
+        b10Comps = []
+        potentialC = None
+        for c in self:
+            if c.getNumberDensity("B10"):
+                potentialC = c
+                b10Comps.append(potentialC)
+                tHot = potentialC.temperatureInC
+                tCold = potentialC.inputTemperatureInC
+                mat = potentialC.material
+        if potentialC is None:
+            return
+
+        # calc volume of boron components
+        area = 0.0
+        for c in b10Comps:
+            if (
+                tHot != c.temperatureInC
+                or tCold != c.inputTemperatureInC
+                or mat != c.material
+            ):
+                runLog.warning(
+                    f"More than 1 component found  in {b.name} with different B10 material"
+                    f" or temperature, using {potentialC} params",
+                    single=True,
+                )
+            area += c.getArea(cold=True)
+
+        volume = area * self.getHeight()
+        if heightHot:
+            self.p.initialB10ComponentVol = (
+                volume / potentialC.getThermalExpansionFactor()
```

Comment:
any concern about volume and linear expansion being 1D? From `component.getThermalExpansionFactor` it looks like this is a linear expansion factor. But volume should be factor^3. (e.g., if working with a unit cube, l, w, h = 1, vol=1. If the linear thermal expansion is 10%, then l,w,h = 1.1 and vol = 1.331).

Reply:
I think the idea here is that `getHeight` above returns the hot block height if this condition is hit. Since the area is cold here, the height is rescaled for this target component to be cold so that the initial volume is stored as cold. 

It might be clearer to have a comment here about this and possibly rescale height first based on this conditional and calculate volume afterwards.

```
if hotHeight:
    # add comment here...
    coldComponentHeight = self.getHeight() / c.getThermalExoansionFactor()
else:
    coldComponentHeight = self.gerHeight()

self.p.initialB10ComponentVol = area * coldComponentHeight
```

Do you agree @onufer / @sammiller11235 ?

Reply:
sure i can tweak. we shouldn't be hitting hot height for any of our internal calcs anyway.